staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -244,6 +268,15 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 		if err != nil {
 			return fmt.Errorf("%s: Failed to list %v: %v", r.name, r.expectedTypeName, err)
 		}
+
+		// For the list with ResourceVersion="0" check if the result was paginated.
+		// If it was, that means that watch cache is disabled and there are a lot of objects,
+		// so force pagination also in the subsequent list calls. Otherwise avoid forcing
+		// pagination to allow listing from watch cache for scalability/performance reasons.
+		if options.ResourceVersion == "0" && paginatedResult {
```

Comment:
why do we need to check options.ResourceVersion, and not just do this:

```go
if paginatedResult {
  r.paginatedResult = true
}
```

Reply:
Actually, the check for RV="0" is needed in my opinion - I extended the comment.